### PR TITLE
syncthing: security update to 1.18.4

### LINKED ISSUE #3691: syncthing: security update to 1.18.4
### CVE IDs

CVE-2021-21404

### Other security advisory IDs

- https://github.com/syncthing/syncthing/security/advisories/GHSA-x462-89pf-6r5h

### Description

See https://nvd.nist.gov/vuln/detail/CVE-2021-21404 .

### Patches

https://github.com/syncthing/syncthing/commit/fb4fdaf4c0a79c22cad000c42ac1394e3ccb6a97

### PoC(s)

N/A

### PR DESCRIPTION
**Pull request by WhiredPlanck**
Topic Description
-----------------

Security update `syncthing`  to 1.18.4

Package(s) Affected
-------------------

`syncthing`

Security Update?
----------------

Yes - Issue Number: #3691 

Test Build(s) Done
------------------

**Primary Architectures**

- [x] AMD64 `amd64`   
- [x] AArch64 `arm64`
    

**Secondary Architectures**

Architectural progress for "secondary," or experimental ports does not impede on merging of this topic.

- [x] Loongson 3 `loongson3`
- [x] PowerPC 64-bit (Little Endian) `ppc64el`

Update(s) Uploaded to Stable
----------------------------

**Primary Architectures**

- [ ] AMD64 `amd64`   
- [ ] AArch64 `arm64`

**Secondary Architectures**

Architectural progress for "secondary," or experimental ports does not impede on merging of this topic.

- [ ] Loongson 3 `loongson3`
- [ ] PowerPC 64-bit (Little Endian) `ppc64el`

### CODE REVIEW

**Comment by WhiredPlanck** [comment]
LGTM on AMD64

image

File: extra-network/syncthing/autobuild/build [hunk 1]
```diff
@@ -1,30 +1,65 @@
-install -d "src/github.com/syncthing"
-mv syncthing "src/github.com/syncthing/syncthing"
+# Adapted from Arch Linux
+abinfo 'Preparing build environment ...'
+install -dv "$SRCDIR"/src/github.com/syncthing
```
Comment (KexyBiscuit) [inline extra-network/syncthing/autobuild/build:3]: Why use `install` here?

Comment (WhiredPlanck) [reply extra-network/syncthing/autobuild/build:3]: Here I just added verbosity based on Arch's PKGBUILD script, should I replace it with `mkdir -pv`?

File: extra-network/syncthing/autobuild/build [hunk 2]
```diff
@@ -1,30 +1,69 @@
...
+export CGO_CXXFLAGS="${CXXFLAGS}"
+if [[ "${CROSS:-$ARCH}" != "loongson3" ]]; then
+    export GOFLAGS="-buildmode=pie -trimpath -mod=readonly -modcacherw"
+else
+    export GOFLAGS="-trimpath -mod=readonly -modcacherw"
+fi
+
+abinfo 'Building binaries ...'
+pushd "$BASEDIR"
 go run build.go -no-upgrade -version v$PKGVER build
```
Comment (MingcongBai) [inline extra-network/syncthing/autobuild/build:22]: Absolute path here.

**Review by MingcongBai — CHANGES_REQUESTED** [review]

**Review by MingcongBai — APPROVED** [review]

**Pull request merged**